app/components/chatbot/AnswerList.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { createRef, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { StyleSheet, View, ScrollView } from "react-native";
+import { TextInput } from "react-native-paper";
+import { RadioButton, Text, TouchableRipple } from "react-native-paper";
+import { SpacerV } from "../../elements/Spacers";
+import { ChatbotAnswer, ChatbotAnswerData } from "../../models/ChatbotData";
+import { chatbotUpdateAnswer } from "../../services/Chatbot";
+
+interface Props {
+  onSubmit: (answer: ChatbotAnswer) => void;
+  answer?: ChatbotAnswer;
+}
+
+const AnswerList = ({
+  onSubmit,
+  answer,
+}: Props) => {
+  const { t } = useTranslation()
+  const scrollViewRef = createRef<ScrollView>()
+  const [value, setValue] = useState("")
+  const [search, setSearch] = useState("")
+  const [items, setItems] = useState<ChatbotAnswerData[]>([])
+
+  useEffect(() => {
+    setValue("")
+    if (answer !== undefined) {
+      setItems(answer.data)
+      if (answer.previousSentValue !== undefined && answer.previousSentValue.length > 0) {
+        itemSelected(JSON.parse(answer.previousSentValue) as ChatbotAnswerData)
+      }
+    }
+    scrollViewRef.current?.scrollTo({x: 0, y: 0, animated: false})
+  }, [answer])
+
+  const itemSelected = (item: ChatbotAnswerData) => {
+    answer?.data.forEach((data) => {
+      data.isSelected = data.key === item.key
+    })
+    setValue(item.key)
+    if (answer !== undefined && item.apiElement !== undefined) {
+      chatbotUpdateAnswer(item.apiElement, answer)
+      onSubmit(answer)
+    }
+  }
+
+  const update = (search: string) => {
+    setSearch(search)
+    if (answer !== undefined) {
+      if (search.length === 0) {
+        setItems(answer.data)
+      } else {
+        setItems(answer.data.filter((item) => { 
+          return item.label.toLowerCase().includes(search.toLocaleLowerCase()) || item.isSelected
+        }))
+      }
+    }
+  }
+
+  return (
+    <View style={styles.container}>
+      {(answer?.data.length ?? 0) >= 10 && (
+        <View>
+          <TextInput
+            inlineImageLeft="search"
+            placeholder={t("kyc.bot.search")}
+            onChangeText={(text) => { 
+              update(text)
+            }}
+            value={search}
+          />
+          <SpacerV />
+        </View>
+      )}
+      <ScrollView ref={scrollViewRef}>
+        {items.map((item) => (
+          <TouchableRipple key={item.key} onPress={() => itemSelected(item)}>
```

Comment:
any specific reason to use TouchableRipple instead of the classic TouchableOpacity?

Reply:
I took it from DeFiDropdown component, which was used initially on the very first chatbot implementation without those pages. What is the difference? Ripple has an additional animated effect, right?

Reply:
ripple is from an external lib (rn-material)
if it doesn't add real value, it's better to use standard native element

Reply:
I would keep it, as other components on the payment side also use it for similar components in this case lists. To have a unified look & feel